internal/stage_clone_repository.go

```
@@ -90,6 +92,54 @@ func testCloneRepository(harness *test_case_harness.TestCaseHarness) error {
+			return fmt.Errorf("CodeCrafters Internal Error: cp failed: %w", err)
+		}
+
+		return nil
+	}()
+
+	defer func() error {
+		if err := os.RemoveAll(gitDir); err != nil {
+			return fmt.Errorf("CodeCrafters Internal Error: delete directory failed: %s", gitDir)
+		}
+		return nil
+	}()
```

golangci-lint (1.64.8)

Error return value is not checked

(errcheck)